app/assets/javascripts/pretty-text/upload-short-url.js.es6

```diff
@@ -97,31 +118,31 @@ function _loadCachedShortUrls($uploads) {
   });
 }
 
-function _loadShortUrls($uploads, ajax) {
+function _loadShortUrls($uploads, ajax, siteSettings) {
   const urls = $uploads.toArray().map(upload => {
     const $upload = $(upload);
     return $upload.data("orig-src") || $upload.data("orig-href");
   });
 
   return lookupUncachedUploadUrls(urls, ajax).then(() =>
-    _loadCachedShortUrls($uploads)
+    _loadCachedShortUrls($uploads, siteSettings)
   );
 }
 
-export function resolveAllShortUrls(ajax) {
+export function resolveAllShortUrls(ajax, siteSettings) {
   const attributes =
     "img[data-orig-src], a[data-orig-href], source[data-orig-src]";
   let $shortUploadUrls = $(attributes);
 
   if ($shortUploadUrls.length > 0) {
-    _loadCachedShortUrls($shortUploadUrls);
+    _loadCachedShortUrls($shortUploadUrls, siteSettings);
 
     $shortUploadUrls = $(attributes);
     if ($shortUploadUrls.length > 0) {
       // this is carefully batched so we can do a leading debounce (trigger right away)
       return debounce(
         null,
-        () => _loadShortUrls($shortUploadUrls, ajax),
+        () => _loadShortUrls($shortUploadUrls, ajax, siteSettings),
```

Comment:
are you sure this debouncing is working as you expect?

Reply:
I am not sure, the debounce was there before. All I added was the siteSettings @jjaffeux

Reply:
I think if you didn't add the debounce it's safe to merge, but maybe a todo for the future @jjaffeux